packages/payment/src/services/payment-module.ts

```diff
@@ -227,12 +239,127 @@ export default class PaymentModuleService implements IPaymentModuleService {
     data: UpdatePaymentDTO[],
     sharedContext?: Context | undefined
   ): Promise<PaymentDTO[]>
-  updatePayment(
-    data: unknown,
-    sharedContext?: unknown
+
+  @InjectTransactionManager("baseRepository_")
+  async updatePayment(
+    data: UpdatePaymentDTO | UpdatePaymentDTO[],
+    @MedusaContext() sharedContext?: Context
   ): Promise<PaymentDTO | PaymentDTO[]> {
-    throw new Error("Method not implemented.")
+    const input = Array.isArray(data) ? data : [data]
+    const result = await this.paymentService_.update(input, sharedContext)
+
+    return await this.baseRepository_.serialize<PaymentDTO[]>(
+      Array.isArray(data) ? result : result[0],
+      {
+        populate: true,
+      }
+    )
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  async capturePayment(
+    data: CreateCaptureDTO,
+    @MedusaContext() sharedContext?: Context
+  ): Promise<PaymentDTO> {
+    let payment = await this.paymentService_.retrieve(
+      data.payment_id,
+      {
+        select: [
+          "amount",
+          "authorized_amount",
+          "captured_amount",
+          "captured_at",
+        ],
+        relations: ["captures"],
+      },
+      sharedContext
+    )
+
+    if (payment.captured_at) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        "The payment is already fully captured."
+      )
+    }
+
+    if (
+      Number(payment.captured_amount) + data.amount >
+      Number(payment.authorized_amount)
+    ) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        "Total captured amount exceeds authorised amount."
+      )
+    }
+
+    await this.paymentService_.capture(data, sharedContext)
+
+    if (
+      Number(payment.captured_amount) + data.amount ===
+      Number(payment.amount)
+    ) {
+      await this.paymentService_.update([
+        { id: payment.id, captured_at: new Date() },
+      ])
+
+      // TODO: set PaymentCollection status if fully captured
+    }
+
+    payment = await this.paymentService_.retrieve(
```

Comment:
I believe we can return the complete entity here. no need to select these fields

Reply:
This will be changed later, this is just temporary until we have an approach for decorating computed properties